### PR TITLE
General code cleanup

### PR DESCRIPTION
**Pull request by johnoneill46**
- Replaced deprecated usages
- Removed duplicate keys in JSON test files
- Removed redundant 'throws Exception' from methods that didn't throw anything
- Simplified assertions (assertTrue(latestGeneralOrder == null); -> assertNull(latestGeneralOrder);)

### CODE REVIEW

File: src/functionalTests/java/uk/gov/hmcts/reform/finrem/functional/caseorchestration/AmendCaseDetailsTest.java [hunk 1]
```diff
@@ -11,8 +11,8 @@
 import uk.gov.hmcts.reform.finrem.functional.IntegrationTestBase;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertThat;
```
Comment (johnoneill46) [inline src/functionalTests/java/uk/gov/hmcts/reform/finrem/functional/caseorchestration/AmendCaseDetailsTest.java:15]: org.junit.Assert.assertThat is now deprecated & should be replaced by hamcrest instance

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/document/DocumentGenerationRequest.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
     @JsonProperty(value = "template", required = true)
     @NotBlank
     private final String template;
-    @ApiModelProperty(value = "Name of the file", required = false)
```
Comment (johnoneill46) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/document/DocumentGenerationRequest.java:20]: Default if unspecified is false so redundant declaration

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/RemoveApplicantDetailsControllerTest.java [hunk 3]
```diff
@@ -36,8 +36,8 @@ public void shouldSuccessfullyRemoveApplicantSolicitorDetails() throws Exception
                 .andExpect(jsonPath("$.data.applicantAddress", is(notNullValue())))
                 .andExpect(jsonPath("$.data.applicantPhone", is("89897876765")))
                 .andExpect(jsonPath("$.data.applicantEmail", is("email01@email.com")))
-                .andExpect(jsonPath("$.data.applicantFMName", is("Test")))
-                .andExpect(jsonPath("$.data.applicantLName", is("Test")))
```
Comment (johnoneill46) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/RemoveApplicantDetailsControllerTest.java:40]: This is the issue with having duplicated keys in the test JSON files - not clear which exactly we were testing

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/integrationtest/AbstractDocumentTest.java [hunk 4]
```diff
@@ -38,7 +37,7 @@
 import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
 import static org.springframework.http.HttpHeaders.AUTHORIZATION;
 import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
-import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
```
Comment (johnoneill46) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/integrationtest/AbstractDocumentTest.java:41]: APPLICATION_JSON_UTF8_VALUE is deprecated and replaced with APPLICATION_JSON_VALUE

**Review by shaunganley — APPROVED** [review]

**Review by HarryH96 — APPROVED** [review]

**Pull request merged**